src/starplot/styles/helpers.py

```diff
@@ -38,9 +41,31 @@ def wrapper(*args, **kwargs):
                 else:
                     kwargs["style"] = style_class(**style)
 
-            elif style is None and style_attr is not None:
-                # if no style overrides and there's a base style, then just pass the base style
-                kwargs["style"] = getattr(args[0].style, style_attr, None)
+            elif style is None:
+                if style_kwargs:
+                    if style_attr is not None:
+                        # if style kwargs are used - build style dict and pass as "style" variable
+                        styling_overrides = {}
+                        for key, value in style_kwargs.items():
+                            parts = key.split("__")[1:]
+                            temp = styling_overrides
+                            for part in parts[:-1]:
+                                temp.setdefault(part, {})
+                                temp = temp[part]
+                            temp[parts[-1]] = value
+                            
+                        base_style = getattr(args[0].style, style_attr).model_dump_json()
+                        base_style = json.loads(base_style)
+
+                        merge_dict(base_style, styling_overrides)
+                        
+                        kwargs["style"] = style_class(**base_style)
+                    else:
+                        kwargs["style"] = style_class(**styling_overrides)
+
+                elif style_attr is not None:
```

Comment:
we need to keep this line as its original:

```
elif style is None and style_attr is not None:
```

This is because in the case that a function is called like this:

`stars(style=ObjectStyle(marker=MarkerStyle(size=2)))`

We want to use the exact style that was passed in, and not the base style from the stylesheet

Reply:
I've made the change to the conditional as you describe, but in your above example call of `stars()`, wouldn't style != None? just wanting to make sure I'm following

Reply:
yeah, style would not be none, but `style_attr` could potentially be not none as well.

In this decorator, style_attr is the style's name in the PlotStyle (passed when creating the plot). That style should always have the lowest priority, and overrides the highest priority. So, if we have this line:

`elif style_attr is not None:`

Then, the call `stars(style=ObjectStyle(marker=MarkerStyle(size=2)))` would end up using the base style and NOT the overrides (because in this case, style is not a dict either, it's an instance of ObjectStyle). But, with the line kept as original:

`elif style is None and style_attr is not None:`

Then, the call `stars(style=ObjectStyle(marker=MarkerStyle(size=2)))` would end up using the override ObjectStyle, which is what we want. In fact, in this case the decorator will keep the style arg as-is.

Hope that makes sense!

Reply:
ah yes, I see where I'd crossed my own wires. makes sense!